implementations/rust/ockam/ockam_api/src/nodes/service.rs

```diff
@@ -209,16 +209,26 @@ impl<'a> NodeManagerProjectsOptions<'a> {
     }
 }
 
+#[derive(Clone)]
+/// Transport to build connection
+pub struct ApiTransport {
+    /// Type of transport being requested
+    pub tt: TransportType,
+    /// Mode of transport being requested
+    pub tm: TransportMode,
+    /// Socket address
+    pub socket_address: Address,
```

Comment:
This should be a `SocketAddr` as discussed at https://github.com/build-trust/ockam/pull/4452/files#r1135198085

Reply:
But if we want to use SocketAddr, then we need to always specify a port. It is a problem, for example in [implementations/rust/ockam/ockam_api/src/util.rs:420](https://github.com/build-trust/ockam/pull/4452/files/ddd74729d9ee3c80c4f64b6a0d64afe7ee56ab7f#diff-fd32a32373b86ee324e19a83006ff012db2a73a22c294a9855b5cfd35dbc1180)

Also, in [implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs:90](https://github.com/build-trust/ockam/pull/4452/files/ddd74729d9ee3c80c4f64b6a0d64afe7ee56ab7f#diff-e399f6f696afc5c1ac63af5380f8ce3a9f0a31191959f5c331ac754fb9ffa036) an error object will get casted to SockAddr, which can throw an error. How should we handle this case?

Reply:
> But if we want to use SocketAddr, then we need to always specify a port. It is a problem, for example in [implementations/rust/ockam/ockam_api/src/util.rs:420](https://github.com/build-trust/ockam/pull/4452/files/ddd74729d9ee3c80c4f64b6a0d64afe7ee56ab7f#diff-fd32a32373b86ee324e19a83006ff012db2a73a22c294a9855b5cfd35dbc1180)

That's a test and that address is probably not used anywhere, that's why it wasn't complaining.

> Also, in [implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs:90](https://github.com/build-trust/ockam/pull/4452/files/ddd74729d9ee3c80c4f64b6a0d64afe7ee56ab7f#diff-e399f6f696afc5c1ac63af5380f8ce3a9f0a31191959f5c331ac754fb9ffa036) an error object will get casted to SockAddr, which can throw an error. How should we handle this case?

That's what we want to avoid. If we have the `socket_addr` typed as a `SocketAddr` in both the command payload and the `NodeManager` registry, then we won't need to do any casting.

Reply:
> That's what we want to avoid. If we have the socket_addr typed as a SocketAddr in both the command payload and the NodeManager registry, then we won't need to do any casting.

Yes, but [implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs:90](https://github.com/build-trust/ockam/pull/4452/files/ddd74729d9ee3c80c4f64b6a0d64afe7ee56ab7f#diff-e399f6f696afc5c1ac63af5380f8ce3a9f0a31191959f5c331ac754fb9ffa036) handles an error case and puts the message into the `sock_addr` field, which can cause a error due to a parsing failure. Should we then use some dummy value in this case instead?

Reply:
OK I see what you mean now. It's so confusing that I didn't see it at first glance. That `socket_address: msg.to_string().into(),` should be replaced by `socket_address: "<none>".into(),`, and the error message either logged as an `error!` or ignored.

Reply:
Okay, got it! But still, `socket_address: "<none>".into()` would throw a parsing error if `socket_address` has a type `SocketAddr`. We could write there "0.0.0.0:0000".into(), and then log the error message, that might work

Reply:
Yeah, that's right, that should do the trick!